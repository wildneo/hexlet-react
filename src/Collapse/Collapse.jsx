/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import cn from 'classnames';

export default class Collapse extends React.Component {
  constructor(props) {
    super(props);
    const { opened } = props;
    this.state = { isShown: opened };

    this.toggle = (event) => {
      event.preventDefault();
      const { isShown } = this.state;
      this.setState({ isShown: !isShown });
    };
  }

  render() {
    const { text } = this.props;
    const { isShown } = this.state;
    return (
      <div>
        <p>
          <a className="btn btn-primary" href="#" onClick={this.toggle}>Link with href</a>
        </p>
        <div className={cn('collapse', { show: isShown })}>
          <div className="card card-body">
            {text}
          </div>
        </div>
      </div>
    );
  }
}

Collapse.defaultProps = {
  opened: true,
  text: null,
};
