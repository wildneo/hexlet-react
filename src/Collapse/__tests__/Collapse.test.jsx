import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Collapse from '../Collapse';

Enzyme.configure({ adapter: new Adapter() });

test('Collapse', () => {
  const wrapper = mount(<Collapse text="hi" opened={false} />);
  expect(wrapper.render()).toMatchSnapshot();

  const button = wrapper.find('a');

  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});

test('Collapse2', () => {
  const wrapper = mount(<Collapse text="i am grut" />);
  expect(wrapper.render()).toMatchSnapshot();
});
