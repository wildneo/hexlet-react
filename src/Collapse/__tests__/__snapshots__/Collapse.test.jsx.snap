// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse 1`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Link with href
    </a>
  </p>
  <div
    class="collapse"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse 2`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Link with href
    </a>
  </p>
  <div
    class="collapse show"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse 3`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Link with href
    </a>
  </p>
  <div
    class="collapse"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse 4`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Link with href
    </a>
  </p>
  <div
    class="collapse show"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse2 1`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Link with href
    </a>
  </p>
  <div
    class="collapse show"
  >
    <div
      class="card card-body"
    >
      i am grut
    </div>
  </div>
</div>
`;
